controller/internal/routes/database_router.go

```diff
@@ -68,27 +89,43 @@ func (r *DatabaseRouter) CreateSnapshot(ae *env.AppEnv, rc *response.RequestCont
 }
 
 func (r *DatabaseRouter) CheckDatastoreIntegrity(ae *env.AppEnv, rc *response.RequestContext, fixErrors bool) {
-	var results []*rest_model.DataIntegrityCheckDetail
-
-	errorHandler := func(err error, fixed bool) {
-		description := err.Error()
-		results = append(results, &rest_model.DataIntegrityCheckDetail{
-			Description: &description,
-			Fixed:       &fixed,
-		})
+	if r.integrityCheck.running.CompareAndSwap(false, true) {
+		r.integrityCheck.fixingErrors = fixErrors
+		go r.runDataIntegrityCheck(ae, fixErrors)
+		rc.Respond(&rest_model.Empty{Data: map[string]interface{}{}, Meta: &rest_model.Meta{}}, http.StatusOK)
```

Comment:
`rc.RespondWithEmptyOk()` is a convenience function.

Reply:
That was supposed to be StatusAccepted, which is why I didn't use the empty ok func. Fixed.